src/solver/qualrout.c

```diff
@@ -330,8 +337,22 @@ void findLinkQual(int i, double tStep)
             c2 = 0.0;
         }
 
-        // --- assign new concen. to link
-        Link[i].newQual[p] = c2;
+	if (Link[i].extPollutFlag[p] == 0)
+	{
+            // --- assign new concen. to link
+            Link[i].newQual[p] = c2;
+	}
+	// --- update mass balance and set external pollutant
+	else if( Link[i].extPollutFlag[p] == 1)
+	{
+	    // --- mass balance update
+            lossExtQual = c2 - Link[i].extQual[p];
+            lossExtQual = lossExtQual * v1/ tStep;
+            massbal_addReactedMass(p, lossExtQual);
+
+            Link[i].newQual[p] = Link[i].extQual[p];
```

Comment:
@abhiramm7 I might be misunderstanding something. I was under the impression that external pollutant was separate from pollutant variables in swmm. But Line 353 is showing that newQual is being updated by extQual. 

Back to the original request, I think it would be really helpful to have docs for the new feature in this PR.

Reply:
For sure, Brooke and I are working on doc sheet. Will add it as soon as we are done.